packages/astro/src/assets/internal.ts

```diff
@@ -79,6 +79,12 @@ export async function getImage(
 			message: AstroErrorData.ExpectedImageOptions.message(JSON.stringify(options)),
 		});
 	}
+	if(typeof options.src === 'undefined') {
+		throw new AstroError({
+			...AstroErrorData.ExpectedImage,
+			message: AstroErrorData.ExpectedImage.message(options.src, 'undefined', JSON.stringify(options))
```

Comment:
This is a bit of a duplicate check from the one in `baseService`, I do think it's okay though because other services should probably have this check as well.